src/runmode-af-packet.c

```diff
@@ -328,20 +300,44 @@ static void *ParseAFPConfig(const char *iface)
         } else if (strcmp(copymodestr, "ips") == 0) {
             SCLogInfo("%s: AF_PACKET IPS mode activated %s->%s", iface, iface, aconf->out_iface);
             aconf->copy_mode = AFP_COPY_MODE_IPS;
-            if (aconf->flags & AFP_TPACKET_V3) {
-                SCLogWarning("%s: using tpacket_v3 in IPS mode will result in high latency", iface);
-            }
         } else if (strcmp(copymodestr, "tap") == 0) {
             SCLogInfo("%s: AF_PACKET TAP mode activated %s->%s", iface, iface, aconf->out_iface);
             aconf->copy_mode = AFP_COPY_MODE_TAP;
-            if (aconf->flags & AFP_TPACKET_V3) {
-                SCLogWarning("%s: using tpacket_v3 in TAP mode will result in high latency", iface);
-            }
         } else {
             SCLogWarning("Invalid 'copy-mode' (not in tap, ips)");
         }
     }
 
+    if (ConfGetChildValueBoolWithDefault(if_root, if_default, "tpacket-v3", &boolval) == 1) {
+        if (boolval) {
+            if (strcasecmp(RunmodeGetActive(), "workers") == 0) {
+#ifdef HAVE_TPACKET_V3
+                SCLogConfig("%s: enabling tpacket v3", aconf->iface);
+                aconf->flags |= AFP_TPACKET_V3;
+#else
+                SCLogWarning("%s: system too old for tpacket v3 switching to v2", iface);
+                aconf->flags &= ~AFP_TPACKET_V3;
+#endif
+            } else {
+                SCLogWarning("%s: tpacket v3 is only implemented for 'workers' runmode."
+                             " Switching to tpacket v2.",
+                        iface);
+                aconf->flags &= ~AFP_TPACKET_V3;
+            }
+        } else {
+            aconf->flags &= ~AFP_TPACKET_V3;
+        }
+    } else if (aconf->copy_mode == AFP_COPY_MODE_NONE) {
+        // If copy mode is none (passive IDS) and "tpacket-v3" is not
+        // present, default to TPACKET_V3.
+        SCLogConfig("%s: enabling tpacket v3", aconf->iface);
+        aconf->flags |= AFP_TPACKET_V3;
```

Comment:
should we do this before checking for `HAVE_TPACKET_V3` above? The system check too old check

Reply:
Good piont. Or should we just require tpacket-v3 and remove these compile time conditionals?

Reply:
Seems like a safe choice as well, lets go with that.